### PR TITLE
Add test for internal/errors/compressor.go

### PR DESCRIPTION
**Pull request by hlts2**
### Description:

I added test case for `internal/errors/compressor.go`.
And I refactored `internal/errors/compressor.go`.

- WHY
Whenever Errorf has no arguments other than a message, nil is returned.
https://github.com/vdaas/vald/blob/master/internal/errors/compressor.go#L40
https://github.com/vdaas/vald/blob/master/internal/errors/compressor.go#L44

- WHAT
Change from `Errorf` to `New`

***grammar check passed***

### Environment:

- Go Version: 1.15.2
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kevindiu — APPROVED** [review]
LGTM

File: internal/errors/compressor.go [hunk 1]
```diff
@@ -18,29 +18,35 @@
...
 		return Errorf("invalid compression level: %d", level)
 	}
 
-	// Compressor
+	// ErrCompressorNameNotFound represents a function to generate an error of compressor not found.
 	ErrCompressorNameNotFound = func(name string) error {
 		return Errorf("compressor %s not found", name)
 	}
 
+	// ErrCompressedDataNotFound returns an error of compressed data not found.
```
Comment (vankichi) [inline internal/errors/compressor.go:31]:
```suggestion
	// ErrCompressedDataNotFound returns an error of compressed data is not found.
```

File: internal/errors/compressor.go [hunk 2]
```diff
@@ -18,29 +18,35 @@
...
 	ErrCompressedDataNotFound = New("compressed data not found")
 
+	// ErrDecompressedDataNotFound returns an error of decompressed data not found.
```
Comment (vankichi) [inline internal/errors/compressor.go:34]:
```suggestion
	// ErrDecompressedDataNotFound returns an error of decompressed data is not found.
```

File: internal/errors/compressor.go [hunk 3]
```diff
@@ -18,29 +18,35 @@
...
 	ErrDecompressedDataNotFound = New("decompressed data not found")
 
+	// ErrCompressFailed returns an error of compress failed.
 	ErrCompressFailed = New("compress failed")
 
+	// ErrDecompressFailed returns an error of decompress failed.
 	ErrDecompressFailed = New("decompress failed")
 
+	// ErrCompressorRegistererIsNotRunning represents a function to generate an error of compressor registerers not running.
```
Comment (vankichi) [inline internal/errors/compressor.go:43]:
```suggestion
	// ErrCompressorRegistererIsNotRunning represents a function to generate an error of compressor registerers is not running.
```

File: internal/errors/compressor_test.go [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+		{
+			name: "returns error when compression leve is 0",
+			args: args{
+				level: 0,
+			},
+			want: want{
+				want: New("invalid compression level: 0"),
+			},
+		},
+	}
```
Comment (vankichi) [inline internal/errors/compressor_test.go:47]: could you add below test pattern?
level is `math.Max(int)`

File: internal/errors/compressor_test.go [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+				level: 100,
+			},
+			want: want{
+				want: New("invalid compression level: 100"),
+			},
+		},
+		{
+			name: "returns error when compression leve is 0",
+			args: args{
+				level: 0,
```
Comment (vankichi) [inline internal/errors/compressor_test.go:41]:
```suggestion
				level: math.Min(int),
```

File: internal/errors/compressor_test.go (hunk 4 above)
Comment (vankichi) [reply internal/errors/compressor_test.go:47]: sorry, `math.Max()` cannot use for `int`.
so how about define it?:
```go
level: int(^uint(0) >> 1)
```

File: internal/errors/compressor_test.go (hunk 5 above)
Comment (vankichi) [reply internal/errors/compressor_test.go:41]: also, could you please define value as minimum number of int?

File: internal/errors/compressor_test.go (hunk 4 above)
Comment (kevindiu) [reply internal/errors/compressor_test.go:47]: you can use math.MaxInt32

File: internal/errors/compressor_test.go (hunk 5 above)
Comment (hlts2) [reply internal/errors/compressor_test.go:42]: I got it

**Review by kevindiu — APPROVED** [review]
LGTM

**Review by vankichi — APPROVED** [review]

**Pull request merged**